runtime/runtime_test.go

```diff
@@ -6122,3 +6122,643 @@ func TestPanics(t *testing.T) {
 	)
 	assert.Error(t, err)
 }
+
+func TestContractUpdateValidation(t *testing.T) {
+
+	t.Parallel()
+
+	runtime := NewInterpreterRuntime()
+
+	newDeployTransaction := func(function, name, code string) []byte {
+		return []byte(fmt.Sprintf(`
+			transaction {
+				prepare(signer: AuthAccount) {
+					signer.contracts.%s(name: "%s", code: "%s".decodeHex())
+				}
+			}`,
+			function,
+			name,
+			hex.EncodeToString([]byte(code)),
+		))
+	}
+
+	accountCode := map[string][]byte{}
+	var events []cadence.Event
+	runtimeInterface := getMockedRuntimeInterfaceForTxUpdate(t, accountCode, events)
+	nextTransactionLocation := newTransactionLocationGenerator()
+
+	deployAndUpdate := func(name string, oldCode string, newCode string) error {
+		deployTx1 := newDeployTransaction(sema.AuthAccountContractsTypeAddFunctionName, name, oldCode)
+		err := runtime.ExecuteTransaction(
+			Script{
+				Source: deployTx1,
+			},
+			Context{
+				Interface: runtimeInterface,
+				Location:  nextTransactionLocation(),
+			},
+		)
+		require.NoError(t, err)
+
+		deployTx2 := newDeployTransaction(sema.AuthAccountContractsTypeUpdateExperimentalFunctionName, name, newCode)
+		err = runtime.ExecuteTransaction(
+			Script{
+				Source: deployTx2,
+			},
+			Context{
+				Interface: runtimeInterface,
+				Location:  nextTransactionLocation(),
+			},
+		)
+		return err
+	}
+
+	t.Run("change field type", func(t *testing.T) {
+		const oldCode = `
+      		pub contract Test1 {
+          		pub var a: String
```

Comment:
Indentation is mixed spaces/tabs here, normalizing to one or the other makes this more readable